prov/sockets/src/sock_ep.c

```diff
@@ -183,6 +187,11 @@ static int sock_ctx_bind_cntr(struct fid *fid, struct fid *bfid, uint64_t flags)
 	struct sock_tx_ctx *tx_ctx;
 	struct sock_rx_ctx *rx_ctx;
 
+	if (flags && !(flags & FI_SEND) && !(flags & FI_RECV) && !(flags & FI_READ) && !(flags & FI_WRITE) && !(flags & FI_REMOTE_READ) && !(flags & FI_REMOTE_WRITE)) {
+		SOCK_LOG_ERROR("Invalid cntr flag\n");
+		return -FI_EINVAL;
+	}
+
```

Comment:
Not sure if this will capture invalid flags. We might just have to create bit masks for CQ and Cntr and check the passed in flags.